File: WebsiteHoTroDichVuBanDoc_Backend/WebsiteHoTroDichVuBanDoc_Backend/app/models/dat_truoc.py
from typing import Optional
from datetime import datetime
from app.models.db_base import DBModel

class DatTruocBase(DBModel):
    maBanSao: int
    maBanDoc: int
    trangThaiDatTruoc: str = 'kichHoat'
    hinhThucThongBao: Optional[str] = None
    thoiDiemHoanThanh: Optional[datetime] = None
    ghiChu: Optional[str] = None

class DatTruocCreate(DatTruocBase):
    pass

class DatTruocUpdate(DBModel):
    trangThaiDatTruoc: Optional[str] = None
    thoiDiemHoanThanh: Optional[datetime] = None
    ghiChu: Optional[str] = None

class DatTruoc(DatTruocBase):
    maDatTruoc: int
    thoiDiemDatTruoc: Optional[datetime] = None

File: WebsiteHoTroDichVuBanDoc/Backend/src/main/java/com/library/website/models/DatTruoc.java
package com.library.website.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "dattruoc", schema = "public")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DatTruoc {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "madattruoc")
    private Long maDatTruoc;

    @ManyToOne
    @JoinColumn(name = "mabansao")
    private BanSao banSao;

    @ManyToOne
    @JoinColumn(name = "mabandoc")
    private BanDoc banDoc;

    @Column(name = "thoidiemdattruoc")
    private LocalDateTime thoiDiemDatTruoc = LocalDateTime.now();

    @Column(name = "trangthaidattruoc")
    private String trangThaiDatTruoc = "kichHoat"; // 'kichHoat', 'daTra', 'daHuy'

    @Column(name = "hinhthucthongbao")
    private String hinhThucThongBao;

    @Column(name = "thoidiemhoanthanh")
    private LocalDateTime thoiDiemHoanThanh;

    @Column(name = "ghichu")
    private String ghiChu;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
